contracts/asset/registry/HomeChainAssetRegistry.sol

```diff
@@ -48,14 +48,13 @@ contract HomeChainAssetRegistry is IAssetRegistry, Ownable {
         require(success, "failed to call register hub");
         
         // set mintable asset
-        try MintableERC20Contract(homeLocation).isMintable() {
-             bytes memory setMintableData = abi.encodeWithSignature(
+        if (ERC20Factory(erc20Factory).mintableList(homeLocation)){
+            bytes memory setMintableData = abi.encodeWithSignature(
                 "setMintable(bytes32)",
                 assetId
             );
             (bool setMintableResult,) = registryHub.call(setMintableData);
```

Comment:
Why not just call setMintable inside TokenFactory when token is created? No need to import TokenFacotry into HomeChainAssetRegistry i think